Lib/csharp/argcargv.i

```diff
@@ -10,9 +10,17 @@
  */
 
 #ifndef SWIGCSHARP_ARGCARGV_PARAM_IDX
-#define SWIGCSHARP_ARGCARGV_PARAM_IDX "0"
+#define SWIGCSHARP_ARGCARGV_PARAM_IDX 0
 #endif SWIGCSHARP_ARGCARGV_PARAM_IDX
+/* This typemap can be defined once
+ * In case you need multiple `argc` indexes,
+ * you can define more typemaps in your project, simply copy these typemaps,
+ * name them as 'ARGV<n>' and use 'SizeParamIndex=n'.
+ */
+
+#define stringify(s) #s
+#define stringifyVal(a) stringify(a)
```

Comment:
What's the purpose of introducing these macros? The naming convention pollutes the global namespace!

Reply:
They're used to stringize a numeric macro value (it's a classic C preprocessor trick for that - if you just apply `#X` to parameter `X` you get `"X"`).

But I really think we want to avoid needing `SWIGCSHARP_ARGCARGV_PARAM_IDX` (or if it's really needed, to have it supplied in a different way such that it doesn't also get wrapped, and ideally so its value can vary between applications without having to copy and modify these typemaps).

Reply:
They convert the `SWIGCSHARP_ARGCARGV_PARAM_IDX` from number to string.
The pre-processing is done in 2 steps:
`stringifyVal(SWIGCSHARP_ARGCARGV_PARAM_IDX)` -> 
`stringify(0)` -> `"0"`

I am open for betters options.
We do no need the macros for any other purpose.